packages/mui-joy/src/Typography/Typography.tsx

```diff
@@ -0,0 +1,275 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { OverridableComponent } from '@mui/types';
+import { unstable_extendSxProp as extendSxProp } from '@mui/system';
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import { TypographyTypeMap, TypographyProps } from './TypographyProps';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
+import { getTypographyUtilityClass } from './typographyClasses';
+
+const useUtilityClasses = (ownerState: TypographyProps) => {
+  const { align, gutterBottom, noWrap, paragraph, variant, level } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      !level && variant,
+      level,
+      align && align !== 'inherit' && `align${capitalize(align)}`,
+      gutterBottom && 'gutterBottom',
+      noWrap && 'noWrap',
+      paragraph && 'paragraph',
+    ],
+  };
+
+  return composeClasses(slots, getTypographyUtilityClass, {});
+};
+
+export const TypographyRoot = styled('span', {
+  name: 'MuiTypography',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      ownerState.variant && styles[ownerState.variant],
+      ownerState.level && styles[ownerState.level],
+      ownerState.align !== 'inherit' && styles[`align${capitalize(ownerState.align)}`],
+      ownerState.noWrap && styles.noWrap,
+      ownerState.gutterBottom && styles.gutterBottom,
+      ownerState.paragraph && styles.paragraph,
+    ];
+  },
+})<{ ownerState: TypographyProps }>(({ theme, ownerState }) => ({
+  margin: 0,
+  // TODO: remove in the next major version
```

Comment:
What next version do these TODOs refer to? As we're building the first version, we can introduce breaking changes (vs material) now, can't we?

Reply:
Yeah, I also doubt my initial thought. @danilo-leal What do you think? should we go with only `level` prop or leave it like this and revisit again before the official release?

Reply:
It doesn't make too much sense to me to maintain `@mui/material` props in any Joy component for the sake of ease of transition. They're meant to be different things with different stakes/trade-offs (at least at some levels), so I'd go for breaking away from Material when appropriate from the get-go.